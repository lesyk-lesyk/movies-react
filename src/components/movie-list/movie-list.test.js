import React from 'react';
import { MovieList } from './index';
import renderer from 'react-test-renderer';

describe('MovieList', () => {
  it('renders movie list', () => {
    const movies = [
      {
        id: '1',
        poster_path: 'https://image.tmdb.org/t/p/w500/6o0UWX2naW7HK45PDNYmoMIk5qs.jpg',
        title: 'No Country for Old Men',
        release_date: '2007',
        genres: ['Crime', 'Drama', 'Thriller']
      }, {
        id: '2',
        poster_path: 'https://image.tmdb.org/t/p/w500/6o0UWX2naW7HK45PDNYmoMIk5qs.jpg',
        title: 'No Country for Old Men',
        release_date: '2007',
        genres: ['Crime', 'Drama', 'Thriller']
      }];

    const tree = renderer
      .create(<MovieList movies={movies} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders default message about empty movie list', () => {
    const movies = [];

    const tree = renderer
      .create(<MovieList movies={movies} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders message from props about empty movie list', () => {
    const movies = [];

    const tree = renderer
      .create(<MovieList movies={movies} message='test message' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
