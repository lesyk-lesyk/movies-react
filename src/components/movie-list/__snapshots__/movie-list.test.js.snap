// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList renders default message about empty movie list 1`] = `
<div
  className="movie-list"
>
  <div
    className="container"
  >
    <div
      className="message"
    >
      No films found
    </div>
  </div>
</div>
`;

exports[`MovieList renders message from props about empty movie list 1`] = `
<div
  className="movie-list"
>
  <div
    className="container"
  >
    <div
      className="message"
    >
      test message
    </div>
  </div>
</div>
`;

exports[`MovieList renders movie list 1`] = `
<div
  className="movie-list"
>
  <div
    className="container"
  >
    <div
      className="movie-box"
    >
      <img
        className="movie-image"
        src="https://image.tmdb.org/t/p/w500/6o0UWX2naW7HK45PDNYmoMIk5qs.jpg"
      />
      <div
        className="movie-info"
      >
        <div
          className="title"
        >
          No Country for Old Men
        </div>
        <div
          className="year"
        >
          2007
        </div>
        <div
          className="genres"
        >
          <span>
            Crime
          </span>
          <span>
            Drama
          </span>
          <span>
            Thriller
          </span>
        </div>
      </div>
    </div>
    <div
      className="movie-box"
    >
      <img
        className="movie-image"
        src="https://image.tmdb.org/t/p/w500/6o0UWX2naW7HK45PDNYmoMIk5qs.jpg"
      />
      <div
        className="movie-info"
      >
        <div
          className="title"
        >
          No Country for Old Men
        </div>
        <div
          className="year"
        >
          2007
        </div>
        <div
          className="genres"
        >
          <span>
            Crime
          </span>
          <span>
            Drama
          </span>
          <span>
            Thriller
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
